File: 0242-valid-anagram/0242-valid-anagram.py
class Solution:
    def isAnagram(self, s: str, t: str) -> bool:
        if len(s) != len(t):
            return False

        char_frequency = {}

        # Build character frequency map for string s
        for char in s:
            char_frequency[char] = char_frequency.get(char, 0) + 1

        # Compare with string t
        for char in t:
            if char not in char_frequency or char_frequency[char] == 0:
                return False
            char_frequency[char] -= 1

        return True

File: 0242-valid-anagram/0242-valid-anagram.cs
using System;
using System.Collections.Generic;

public class Solution {
    public bool IsAnagram(string s, string t) {
        if (s.Length != t.Length) return false;
        int[] mapp = new int[26];
        for (int i=0; i<s.Length; ++i) {
            mapp[s[i] - 'a']++;
            mapp[t[i] - 'a']--;
        }
        for (int i=0; i<26; ++i) {
            if (mapp[i] != 0) return false;
        }
        return true;
    }
}
